File: 3959-check-good-integer/3959-check-good-integer.java
class Solution {
    public boolean checkGoodInteger(int n) {
        int sum = 0;
        int sqSum = 0;

        while (n > 0) {
            int digit = n % 10;
            sum += digit;
            sqSum += digit * digit;
            n /= 10;
        }

        return sqSum - sum >= 50;
    }
}

File: 3959-check-good-integer/3959-check-good-integer.kt
class Solution {
    fun checkGoodInteger(n: Int): Boolean {
        var num = n
        var ds = 0
        var ss = 0

        while (num > 0) {
            val curr = num % 10
            ds += curr
            ss += curr * curr
            num /= 10
        }

        return ss - ds >= 50
    }
}